File: LeetCode/12IntegerToRoman/IntegerToRoman.cpp
class Solution {
public:
    string intToRoman(int num) {
        // Single pass -> Time:O(n), Space:O(1)
        vector<int> int_vect {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        vector<string> char_vect {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        string ans = "";
        int count;
        for (int i = 0; i < int_vect.size(); i++) {
            count = num / int_vect[i];
            for (int j = 0; j < count; j++) {
                ans += char_vect[i];
            }
            num %= int_vect[i];
        }
        return ans;
    }
};

File: LeetCode/12IntegerToRoman/IntegerToRoman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        # Single pass -> Time:O(n), Space:O(1)
        symbol_value = [
            [1000, 'M'],
            [900, 'CM'],
            [500, 'D'],
            [400, 'CD'],
            [100, 'C'],
            [90, 'XC'],
            [50, 'L'],
            [40, 'XL'],
            [10, 'X'],
            [9, 'IX'],
            [5, 'V'],
            [4, 'IV'],
            [1, 'I'],
        ]
        ans = ""
        for val, sym in symbol_value:
            ans += sym*(num//val)
            num %= val
        return ans
